src/Components/Header/Header.jsx

```diff
@@ -1,7 +1,28 @@
 import { Link } from "react-router-dom";
+import {useState} from "react";
+import { useNavigate } from "react-router";
+import { useAuth } from "../../Context/AuthContext/auth-context";
 import styles from "./Header.module.css";
 
 const Header = () => {
+    const {authState,authDispatch} = useAuth();
+    const userName = authState.user;
+    const redirect = useNavigate();
+
+    const checkStatus = (userName) => {
+        return userName ? "Logout" : "Login";
+    }
+
+    const logoutHandler = () => {
+        redirect("/");
+        localStorage.removeItem("token");
+        localStorage.removeItem("user");
+        authDispatch({type:"LOGOUT"});
+    }
+
+    const userHandler = async (type) => {
+        type === "Login" ? redirect("/login"): logoutHandler();
+    }
```

Comment:
Hey @Tejinder15 can you just tell is there any specific reason of using async here?